app/models/landing_page.rb

```diff
@@ -7,21 +7,28 @@ def initialize(content_store_response)
     if content_store_response.dig("details", "blocks")
       super(content_store_response)
     else
-      super(content_store_response.deep_merge(load_additional_content(content_store_response["base_path"])))
+      super(
+        content_store_response,
+        override_content_store_hash: load_additional_content(content_store_response)
+      )
     end
 
-    @blocks = (@content_store_response.dig("details", "blocks") || []).map { |block_hash| BlockFactory.build(block_hash) }
+    @blocks = (content_store_hash.dig("details", "blocks") || []).map { |block_hash| BlockFactory.build(block_hash) }
   end
 
 private
 
   # SCAFFOLDING: can be removed (and reference above) when full content items
   # including block details are available from content-store
-  def load_additional_content(base_path)
+  def load_additional_content(content_store_response)
+    base_path = content_store_response["base_path"]
     file_slug = base_path.split("/").last.gsub("-", "_")
     filename = Rails.root.join("#{ADDITIONAL_CONTENT_PATH}/#{file_slug}.yaml")
-    return { "details" => {} } unless File.exist?(filename)
+    content_hash = content_store_response.to_hash
+    return content_hash unless File.exist?(filename)
```

Comment:
Moving the logic changes to this method is a much more elegant solution. 🎉 
If I wanted to be really pedantic I would say that you could switch these to lines around and return nil if a file doesn't exist. But that's really minor and not blocker.